File: TestDagger2/src/main/java/someday/com/testdagger2/MainActivity.java
package someday.com.testdagger2;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity
{

    @Inject
    ClassA mClassA;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerClassComponent.create().inject(this);
        mClassA.toClassB();
    }
}

File: TestDagger2Kt/src/main/java/someday/com/testdagger2kt/nomvp/MainActivity.kt
package someday.com.testdagger2kt.nomvp

import android.os.Bundle
import android.util.Log
import someday.com.testdagger2kt.R
import javax.inject.Inject

class MainActivity : BaseActivity()
{
    @Inject
    lateinit var people: People
    @Inject
    lateinit var people2: People2
    @Inject
    lateinit var mMainFragment: MainFragment

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        people.hello();
        Log.d("ssssss",((null == mMainFragment).toString()))
        people2.hello();
    }


}
